packages/framework/src/Model/Product/ProductService.php

```diff
@@ -33,29 +33,29 @@ class ProductService
     private $pricingSetting;
 
     /**
-     * @var \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPriceRecalculationScheduler
+     * @var \Shopsys\FrameworkBundle\Model\Product\ProductCategoryDomainFactoryInterface
      */
-    private $productPriceRecalculationScheduler;
+    protected $productCategoryDomainFactory;
 
     /**
-     * @var \Shopsys\FrameworkBundle\Model\Product\ProductCategoryDomainFactoryInterface
+     * @var \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPriceRecalculator
      */
-    protected $productCategoryDomainFactory;
+    protected $productPriceRecalculator;
 
     public function __construct(
         ProductPriceCalculation $productPriceCalculation,
         InputPriceCalculation $inputPriceCalculation,
         BasePriceCalculation $basePriceCalculation,
         PricingSetting $pricingSetting,
-        ProductPriceRecalculationScheduler $productPriceRecalculationScheduler,
-        ProductCategoryDomainFactoryInterface $productCategoryDomainFactory
+        ProductCategoryDomainFactoryInterface $productCategoryDomainFactory,
+        ProductPriceRecalculator $productPriceRecalculator
     ) {
         $this->productPriceCalculation = $productPriceCalculation;
         $this->inputPriceCalculation = $inputPriceCalculation;
         $this->basePriceCalculation = $basePriceCalculation;
         $this->pricingSetting = $pricingSetting;
-        $this->productPriceRecalculationScheduler = $productPriceRecalculationScheduler;
         $this->productCategoryDomainFactory = $productCategoryDomainFactory;
+        $this->productPriceRecalculator = $productPriceRecalculator;
```

Comment:
Service should not be dependent on any Database using Class like Recalculator. It is because services should be easily tested so you do not have to mock every dependency.